### PR TITLE
Feature:support column level foreign key in mysql mode,issue 482

### PR DESCRIPTION
**Pull request by LJYer**
### What changes were proposed in this pull request?

This changes is proposed for issue #482  in oceanbase competition to support column level foreign key in mysql mode

### Why are the changes needed?

support new feature: column level foreign key in mysql mode

### Will break the compatibility? How if so?

No

### Does this PR introduce any user-facing change?

No

### Checklist

- [√] I've run the tests to see all new and existing tests pass.
- [√] If this Pull Request resolves an issue, I linked to the issue in the text above.
- [√] I've informed the technical writer about the documentation change if necessary.

### CODE REVIEW

File: src/sql/resolver/ddl/ob_ddl_resolver.cpp [hunk 1]
```diff
@@ -5647,32 +5647,61 @@ int ObDDLResolver::resolve_foreign_key_node(
...
-        LOG_WARN("failed to resolve foreign key parent columns", K(ret));
-      } else if (OB_FAIL(resolve_foreign_key_options(reference_options, arg.update_action_, arg.delete_action_))) {
-        LOG_WARN("failed to resolve foreign key options", K(ret));
-      } else if (OB_FAIL(resolve_foreign_key_name(constraint_name, arg.foreign_key_name_))) {
-        LOG_WARN("failed to resolve foreign key name", K(ret));
-      } else if (OB_FAIL(check_foreign_key_reference(arg, is_alter_table, NULL))) {
-        LOG_WARN("failed to check reference columns", K(ret));
-      } else if (OB_FAIL(resolve_match_options(match_options))) {
-        LOG_WARN("failed to resolve match options", K(ret));
+    if (NULL == column) {  // table level foreign key in mysql mode
```
Comment (liboyang0730) [inline src/sql/resolver/ddl/ob_ddl_resolver.cpp:5650]: 代码看上去没啥问题，但是需要加测试用例。

Comment (zhanhaozhao) [reply src/sql/resolver/ddl/ob_ddl_resolver.cpp:5650]: 请问加测试用例是给@LINxiansheng的回复吗？我们比赛还有一天就结束了，辛苦帮忙处理下，谢谢！

Comment (zhanhaozhao) [reply src/sql/resolver/ddl/ob_ddl_resolver.cpp:5650]: 请问加测试用例是给@LINxiansheng的回复吗？我们比赛还有一天就结束了，辛苦帮忙处理下，谢谢！

Comment (LJYer) [reply src/sql/resolver/ddl/ob_ddl_resolver.cpp:5648]: 已经增加测试用例了，麻烦您看一下

**Comment by LINxiansheng** [comment]
Thanks for your contribution but we have not updated the code for a long time.
Close it.

**Pull request closed without merging**